assets/js/base/components/cart-checkout/shipping-rates-control-package/index.tsx

```diff
@@ -66,13 +70,25 @@ export const ShippingRatesControlPackage = ( {
 	showItems = false,
 }: PackageProps ): ReactElement => {
 	const { selectShippingRate } = useSelectShippingRate();
+	const data = __experimentalApplyCheckoutFilter( {
```

Comment:
@nielslange, why do we need the `__experimentalApplyCheckoutFilter` filter? Are going to run into some issues by using directly `packageData.name` value?

Reply:
> @nielslange, why do we need the __experimentalApplyCheckoutFilter filter?

I was using `__experimentalApplyCheckoutFilter` as we're using that filter in other files as well. AFAIK, this is the only filter we're currently using for adding filters to the checkout. 

> Are going to run into some issues by using directly packageData.name value?

Would you mind explaining to me what you mean by that, @tarhi-saad?

Reply:
> Would you mind explaining to me what you mean by that, @tarhi-saad?

@nielslange, I was just wondering why we need to apply a filter to get the package name in this instance, knowing that on the initial code we didn't have to:

1. Previous code (getting package name directly from the `packageData` props):

```jsx
<div className="wc-block-components-shipping-rates-control__package-title">
  { packageData.name } // ==> This line
</div>
```

2. Current code (Getting the package name through a filter):

```jsx
const data = __experimentalApplyCheckoutFilter( { // ===> The filter
  filterName: 'itemName',
  defaultValue: packageData.name,
} );

<div
  className="wc-block-components-shipping-rates-control__package-title"
  dangerouslySetInnerHTML={ { __html: data } } // ==> The same package name
/>
```

Is there a specific reason why we are using the `__experimentalApplyCheckoutFilter` filter?

Reply:
Ah, that's what you mean. I used the `__experimentalApplyCheckoutFilter` filter based on @opr [comment](https://github.com/woocommerce/woocommerce-blocks/issues/6903#issuecomment-1227105090). I should have double-checked why we want to run the input through a filter. 

@opr Would you mind clarifying what you had in mind [in this comment](https://github.com/woocommerce/woocommerce-blocks/issues/6903#issuecomment-1227105090)? Did you intend to track the usage of the checkout filter by running the input through a filter?

Reply:
[The `itemName` filter](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/third-party-developers/extensibility/checkout-block/available-filters.md#cart-line-items) is used to modify the item name to make it different in the Cart/Checkout blocks than it is in the back end. This was required by 3991-gh-woocommerce/woocommerce-subscriptions and added in #3807

What is `packageData.name` here, is it the item name? If not then it should not go through that specific filter, only item names should.

Reply:
> [The `itemName` filter](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/third-party-developers/extensibility/checkout-block/available-filters.md#cart-line-items) is used to modify the item name to make it different in the Cart/Checkout blocks than it is in the back end. This was required by 3991-gh-woocommerce/woocommerce-subscriptions and added in #3807
> 
> What is `packageData.name` here, is it the item name? If not then it should not go through that specific filter, only item names should.

Thanks for clarifying this, @opr. `packageData.name` is not the name of the cart item, but the name of the shipping package. Therefore, I removed the `__experimentalApplyCheckoutFilter`.

@tarhi-saad Goo that you caught this one! Would you mind doing a final review of this PR?